### ISSUE TITLE
Test new DLAC and LSAC architectures

### ISSUE THREAD

**Issue by rickstaa** [rickstaa/stable-learning-control#36]
In this issue, the results of two new architectures DLAC and LSAC are compared with the original LAC algorithm. To do this I will use the oscillator environment. I will also set the Environment and Algorithm seeds to 0.

## LAC results

The original LAC algorithm gives the following results:

**Good policy**
![image](https://user-images.githubusercontent.com/17570430/97327879-b6ad8f00-1875-11eb-9f7f-815bbda4e1ad.png)

**Bad policy**
![image](https://user-images.githubusercontent.com/17570430/97327932-c4fbab00-1875-11eb-84e8-ba7a42c8dc8d.png)

## DLAC results

In the double-Lypaunov Actor-Critic, two Lyapunov critics are used instead of one. Following the maximum L, value is used for calculating the actor loss. This is similar to the double-Q trick that is used in the original SAC algorithm.

### Results

In the current from the double-Lyapunov Soft actor, the critic is not able to train. I, however, think this is due to an error in the implementation. I will postpone researching this architecture after the Pytorch version is fully ready as in there it is easier to debug.

## LSAC

In the Lyapunov Soft Actor-Critic (Couldn't think of a name) contains both a Lyapunov critic and a normal soft critic. Following the results of both these critics are combined in the loss function for the policy:

![image](https://user-images.githubusercontent.com/17570430/97293257-b8af2800-184c-11eb-98d8-3214b4ed6ce0.png)
